test/integration/test/custom_vocabularies.test.js

```diff
@@ -0,0 +1,32 @@
+const clientHelper = require('../src/client-helper');
+const configHelper = require('../src/config-helper');
+const CustomVocabularyStatus = require('../../../dist/src/models/CustomVocabularyStatus').CustomVocabularyStatus;
+const client = clientHelper.getCustomVocabulariesClient(configHelper.getApiKey());
+
+test('Can submit custom vocabulary', async () => {
+    const informationSubmit = await client.submitCustomVocabularies([{phrases:['some','custom','vocabularies']}]);
+
+    expect(informationSubmit.status).toBe(CustomVocabularyStatus.InProgress);
+    expect(informationSubmit.created_on).not.toBeNull();
+    expect(informationSubmit.id).not.toBeNull();
+    expect(informationSubmit.failure).toBeUndefined();
+});
+
+test('Can retreive submitted custom vocabulary', async () => {
+    jest.useRealTimers();
+
+    const informationSubmit = await client.submitCustomVocabularies([{phrases:['some','custom','vocabularies']}]);
+
+    let customVocabularyInformation = await client.getCustomVocabularyInformation(informationSubmit.id);
+
+    while(customVocabularyInformation.status == CustomVocabularyStatus.InProgress)
+    {
+        await new Promise(resolve => setTimeout(resolve, 5000));
+        customVocabularyInformation = await client.getCustomVocabularyInformation(informationSubmit.id);
+    }
+
+    expect(customVocabularyInformation.status).toBe(CustomVocabularyStatus.Complete);
+    expect(information.created_on).toBe(informationSubmit.created_on);
+    expect(information.id).toBe(informationSubmit.id);
+    expect(information.failure).toBeUndefined();
```

Comment:
From my experience this won't work because jest will lock and doesn't play nice with while loops. Because this is an async test you will need to wrap your `expect` assertions in an interval. 

It should look something like this:
```
var intervalObject = setInterval(function(){
        (async () => {
            customVocabularyInformation = await client.getCustomVocabularyInformation(informationSubmit.id);
            if (customVocabularyInformation.status === CustomVocabularyStatus.Complete) {
                expect(customVocabularyInformation.status).toBe(CustomVocabularyStatus.Complete);
                expect(customVocabularyInformation.created_on).toBe(informationSubmit.created_on);
                expect(customVocabularyInformation.id).toBe(informationSubmit.id);
                expect(customVocabularyInformation.failure).toBeUndefined();
                clearInterval(intervalObject);
                done();
            }
        })()
    }, 15000);
```

Note that I've added the done() so that the test will know when it's finished

Because this test will most likely run over the default test timeout of 5 secs you will need to increase the amount of time jest waits for this test to finish. Look at my last comment.

Reply:
Thanks a lot Chance!